js/packages/teams-ai/src/AzureOpenAIModerator.ts

```diff
@@ -0,0 +1,222 @@
+/* eslint-disable security/detect-object-injection */
+/**
+ * @module teams-ai
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Plan, PredictedDoCommand, PredictedSayCommand } from './Planner';
+import { TurnState } from './TurnState';
+import { DefaultTurnState } from './DefaultTurnStateManager';
+import { TurnContext } from 'botbuilder';
+import {
+    CreateModerationResponseResultsInner,
+    AzureOpenAIClient,
+    AzureOpenAIModeratorCategory,
+    CreateContentSafetyResponse,
+    OpenAIClientResponse,
+    ModerationResponse
+} from './OpenAIClients';
+import { PromptTemplate } from './Prompts';
+import { ConfiguredAIOptions, AI } from './AI';
+import { OpenAIModerator, OpenAIModeratorOptions } from './OpenAIModerator';
+
+/**
+ * Options for the OpenAI based moderator.
+ */
+export interface AzureOpenAIModeratorOptions extends OpenAIModeratorOptions {
+    /**
+     * Optional: Azure OpenAI Content Safety Categories
+     * Allowed values: "Hate", "Sexual", "SelfHarm", "Violence"
+     */
+    categories?: AzureOpenAIModeratorCategory[];
+}
+
+/**
+ * A moderator tha users OpenAI's moderation API to review prompts and plans for safety.
```

Comment:
nit: A moderator "that uses"

Reply:
good catch. I believe it was actually a GitHub Copilot who wrote this comment. :)
I'll fix it and recommit